src/it/process-jar/pom.xml

```diff
@@ -0,0 +1,116 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>org.jenkins-ci.plugins</groupId>
+    <artifactId>plugin</artifactId>
+    <version>1.424</version>
+  </parent>
+
+  <groupId>org.jenkins-ci.tools.hpi.its</groupId>
+  <artifactId>process-jar</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <packaging>hpi</packaging>
+
+  <name>MyNewPlugin</name>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-databind</artifactId>
+      <version>2.6.3</version>
+      <!--TODO we mark it as optional to exclude from the WEB-INF/lib but when shade is fixed need to verify that it shades in these optional dependencies-->
+      <optional>true</optional>
```

Comment:
@jglick wrt #18 this marks the jackson dependencies as excluded from the .hpi file... in fact they will not be on the classpath at all... now there are a couple of issues with the current shade plugin:

* it cannot process a side artifact, it will only process the main artifact
* I am unsure if it processes optional artifacts (if it does then that should provide a workaround for @KostyaSha's issue in https://github.com/jenkinsci/maven-hpi-plugin/pull/18#issuecomment-126306038 as he can then just mark the dependency as `<optional>true</optional>` and stop trying to create a dependency reduced pom)

The two above can probably be fixed by me in the shade plugin...

For #18, I think that was assuming that you would use the maven-jar-plugin to generate the `jar` (which would then be the main artifact and then maven will blow up in your face when the hpi plugin tries to replace the main artifact with the hpi file... never mind that the .jar file resolved from the repository wouldn't be the shaded one at all)

So in short I think #18 was misguided and should probably be dropped. /cc @oleg-nenashev